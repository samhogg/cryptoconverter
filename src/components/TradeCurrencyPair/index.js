import React from "react";
import PropTypes from "prop-types";
import {
  TradeCurrencyPairContainer,
  TradeLabel,
  TradeInput
} from "./styledComponents";

const TradeCurrencyPair = ({ currency, value, onChange }) => (
  <TradeCurrencyPairContainer>
    <TradeLabel htmlFor={`currency-input-${currency}`}>{currency}</TradeLabel>
    <TradeInput
      id={`currency-input-${currency}`}
      placeholder={currency}
      value={value || ""}
      onChange={onChange}
      disabled={!onChange}
      readOnly={!onChange}
    />
  </TradeCurrencyPairContainer>
);

TradeCurrencyPair.propTypes = {
  currency: PropTypes.string.isRequired,
  value: PropTypes.number,
  onChange: PropTypes.func
};

export default TradeCurrencyPair;
