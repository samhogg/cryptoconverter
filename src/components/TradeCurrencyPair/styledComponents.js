import styled from "styled-components";

const TradeCurrencyPairContainer = styled.div`
  display: flex;
  background-color: white;
  border-radius: 3px;
  overflow: hidden;
  border: 2px solid #ccc;
`;
const TradeLabel = styled.label`
  display: block;
  padding: 0.5rem;
  width: 3rem;
  background-color: #ccc;
  text-align: center;
  color: #666;
`;
const TradeInput = styled.input`
  box-sizing: border-box;
  flex: 1;
  padding: 0.5rem;
  border: 0;
`;

export { TradeCurrencyPairContainer, TradeLabel, TradeInput };
