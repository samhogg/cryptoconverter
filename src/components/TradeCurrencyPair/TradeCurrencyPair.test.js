import React from "react";
import TradeCurrencyPair from ".";
import renderer from "react-test-renderer";

it("Renders with zero balances", () => {
  const tree = renderer
    .create(<TradeCurrencyPair currency="USD" value={0} onChange={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders with a non zero balance", () => {
  const tree = renderer
    .create(
      <TradeCurrencyPair currency="USD" value={1000} onChange={() => {}} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders without an onChange handler", () => {
  const tree = renderer
    .create(<TradeCurrencyPair currency="USD" value={1000} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
