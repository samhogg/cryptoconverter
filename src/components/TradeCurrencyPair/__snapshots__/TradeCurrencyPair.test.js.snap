// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with a non zero balance 1`] = `
<div
  className="styledComponents__TradeCurrencyPairContainer-euJuRu eFLMYC"
>
  <label
    className="styledComponents__TradeLabel-izuSoq dTyLVH"
    htmlFor="currency-input-USD"
  >
    USD
  </label>
  <input
    className="styledComponents__TradeInput-eaQxiw kFzhqY"
    disabled={false}
    id="currency-input-USD"
    onChange={[Function]}
    placeholder="USD"
    readOnly={false}
    value={1000}
  />
</div>
`;

exports[`Renders with zero balances 1`] = `
<div
  className="styledComponents__TradeCurrencyPairContainer-euJuRu eFLMYC"
>
  <label
    className="styledComponents__TradeLabel-izuSoq dTyLVH"
    htmlFor="currency-input-USD"
  >
    USD
  </label>
  <input
    className="styledComponents__TradeInput-eaQxiw kFzhqY"
    disabled={false}
    id="currency-input-USD"
    onChange={[Function]}
    placeholder="USD"
    readOnly={false}
    value=""
  />
</div>
`;

exports[`Renders without an onChange handler 1`] = `
<div
  className="styledComponents__TradeCurrencyPairContainer-euJuRu eFLMYC"
>
  <label
    className="styledComponents__TradeLabel-izuSoq dTyLVH"
    htmlFor="currency-input-USD"
  >
    USD
  </label>
  <input
    className="styledComponents__TradeInput-eaQxiw kFzhqY"
    disabled={true}
    id="currency-input-USD"
    onChange={undefined}
    placeholder="USD"
    readOnly={true}
    value={1000}
  />
</div>
`;
